### PR TITLE
Implemented saving statistics

### PR DESCRIPTION
**Pull request by Laugslander**

### CODE REVIEW

File: Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/RoomWrapper.java [hunk 1]
```diff
@@ -77,16 +81,23 @@ public void join(PlayerWrapper player, String password) {
...
             PlayerLeftSessionMessage message = new PlayerLeftSessionMessage(player.getPlayerId());
 
             players.remove(player);
             players.stream()
                     .map(PlayerWrapper::getConnection)
                     .forEach(c -> c.send(message));
+
+            if (players.isEmpty()) {
+                room.setHost(null);
+            } else if (room.getHost().equals(unwrapped) && players.size() > 0) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/RoomWrapper.java:96]: MAJOR Use isEmpty() to check whether the collection is empty or not. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1155)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/RoomWrapper.java [hunk 2]
```diff
@@ -77,16 +81,23 @@ public void join(PlayerWrapper player, String password) {
     }
 
     public void leave(PlayerWrapper player) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/RoomWrapper.java:83]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerRmiController.java [hunk 3]
```diff
@@ -89,6 +90,60 @@ public void changeHost(String roomName, String host) {
...
+        try {
+            privilege = mainServerForGameServer.getPrivilege(username);
+        } catch (RemoteException e) {
+            LOGGER.log(Level.WARNING, "An error occurred while retrieving the privilege of a user.", e);
+        }
+
+        return privilege;
+    }
+
     public void close() {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerRmiController.java:147]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerRmiController.java [hunk 4]
```diff
@@ -89,6 +90,60 @@ public void changeHost(String roomName, String host) {
...
 
+    public void addGoals(String username, int goals) {
+        try {
+            mainServerForGameServer.addGoals(username, goals);
+        } catch (RemoteException e) {
+            LOGGER.log(Level.WARNING, "An error occurred while adding goals.", e);
+        }
+    }
+
+    public void addAssists(String username, int assists) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerRmiController.java:101]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerRmiController.java [hunk 5]
```diff
@@ -89,6 +90,60 @@ public void changeHost(String roomName, String host) {
...
+    public void incrementGamesWon(String username) {
+        try {
+            mainServerForGameServer.incrementGamesWon(username);
+            incrementGamesPlayed(username);
+        } catch (RemoteException e) {
+            LOGGER.log(Level.WARNING, "An error occurred while incrementing games won.", e);
+        }
+    }
+
+    public void incrementGamesLost(String username) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerRmiController.java:118]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerRmiController.java [hunk 6]
```diff
@@ -89,6 +90,60 @@ public void changeHost(String roomName, String host) {
...
+
+    public void incrementGamesPlayed(String username) {
+        try {
+            mainServerForGameServer.incrementGamesPlayed(username);
+        } catch (RemoteException e) {
+            LOGGER.log(Level.WARNING, "An error occurred while incrementing games played.", e);
+        }
+    }
+
+    public Privilege getPrivilege(String username) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerRmiController.java:135]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerRmiController.java [hunk 7]
```diff
@@ -89,6 +90,60 @@ public void changeHost(String roomName, String host) {
         }
     }
 
+    public void addGoals(String username, int goals) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerRmiController.java:93]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerRmiController.java [hunk 8]
```diff
@@ -89,6 +90,60 @@ public void changeHost(String roomName, String host) {
...
+
+    public void addAssists(String username, int assists) {
+        try {
+            mainServerForGameServer.addAssists(username, assists);
+        } catch (RemoteException e) {
+            LOGGER.log(Level.WARNING, "An error occurred while adding assists.", e);
+        }
+    }
+
+    public void incrementGamesWon(String username) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerRmiController.java:109]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerRmiController.java [hunk 9]
```diff
@@ -89,6 +90,60 @@ public void changeHost(String roomName, String host) {
...
+    public void incrementGamesLost(String username) {
+        try {
+            mainServerForGameServer.incrementGamesLost(username);
+            incrementGamesPlayed(username);
+        } catch (RemoteException e) {
+            LOGGER.log(Level.WARNING, "An error occurred while incrementing games lost.", e);
+        }
+    }
+
+    public void incrementGamesPlayed(String username) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerRmiController.java:127]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

**Comment by ghost** [comment]
SonarQube analysis reported 30 issues
* MAJOR 2 major
* MINOR 28 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [GameServerRmiController.java#L39](https://github.com/PTS3-S34A/Game-Server/blob/6b4bbc59f7867929d1dcd48ce302175c8978fa47/Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerRmiController.java#L39): Refactor this method to throw at most one checked exception instead of: java.rmi.NotBoundException, java.rmi.RemoteException [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1160)
1. MINOR [GameServerRmiController.java#L30](https://github.com/PTS3-S34A/Game-Server/blob/6b4bbc59f7867929d1dcd48ce302175c8978fa47/Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerRmiController.java#L30): Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [GameServerRmiController.java#L35](https://github.com/PTS3-S34A/Game-Server/blob/6b4bbc59f7867929d1dcd48ce302175c8978fa47/Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerRmiController.java#L35): Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [GameServerRmiController.java#L39](https://github.com/PTS3-S34A/Game-Server/blob/6b4bbc59f7867929d1dcd48ce302175c8978fa47/Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerRmiController.java#L39): Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [GameServerRmiController.java#L43](https://github.com/PTS3-S34A/Game-Server/blob/6b4bbc59f7867929d1dcd48ce302175c8978fa47/Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerRmiController.java#L43): Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [GameServerRmiController.java#L48](https://github.com/PTS3-S34A/Game-Server/blob/6b4bbc59f7867929d1dcd48ce302175c8978fa47/Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerRmiController.java#L48): Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [GameServerRmiController.java#L61](https://github.com/PTS3-S34A/Game-Server/blob/6b4bbc59f7867929d1dcd48ce302175c8978fa47/Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerRmiController.java#L61): Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [GameServerRmiController.java#L69](https://github.com/PTS3-S34A/Game-Server/blob/6b4bbc59f7867929d1dcd48ce302175c8978fa47/Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerRmiController.java#L69): Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [GameServerRmiController.java#L77](https://github.com/PTS3-S34A/Game-Server/blob/6b4bbc59f7867929d1dcd48ce302175c8978fa47/Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerRmiController.java#L77): Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [GameServerRmiController.java#L85](https://github.com/PTS3-S34A/Game-Server/blob/6b4bbc59f7867929d1dcd48ce302175c8978fa47/Soccar [Game-Server]/src/nl/soccar/gameserver/controller/rmi/GameServerRmiController.java#L85): Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

**Review by BeeHiveJava — APPROVED** [review]

**Pull request merged**